app/javascript/components/textual_summary/table_list_view.jsx

```diff
@@ -3,40 +3,25 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import MiqStructuredList from '../miq-structured-list';
 
-const simpleRow = (row, i, colOrder) => (
-  <tr key={i} className="no-hover">
-    {colOrder.map((col, j) => <td key={j}>{row[col]}</td>)}
-  </tr>
-);
-
-const clickableRow = (row, i, colOrder, rowLabel, onClick) => (
-  <tr key={i}>
-    {colOrder.map((col, j) => (
-      <td key={j} title={rowLabel}>
-        <a href={row.link} onClick={(e) => onClick(row, e)}>
-          {`${row[col]}`}
-        </a>
-      </td>
-    ))}
-  </tr>
-);
-
-const renderRow = (row, i, colOrder, rowLabel, onClick) => (
-  row.link ? clickableRow(row, i, colOrder, rowLabel, onClick) : simpleRow(row, i, colOrder)
-);
-
 export default function TableListView(props) {
   const { headers, values, title } = props;
 
   /** Function to generate rows for structured list. */
   const miqListRows = (list) => {
-    const headerKeys = props.headers.map((item) => (item.key));
-    return list.map((item) => (headerKeys.map((header) => item[header])));
+    // For More Complicated Tables, e.g. Tenant Quotas
+    if (!!headers[0].key) {
+      const headerKeys = headers.map((item) => (item.key));
+      return list.map((item) => (headerKeys.map((header) => item[header])));
+    }
+
+    const data = [];
+    list.map((item) => data.push({ ...item, label: item.name }));
+    return data;
   };
 
   return (
     <MiqStructuredList
-      headers={headers.map((item) => item.label)}
+      headers={!!headers[0].key ? headers.map((item) => item.label) : headers}
```

Comment:
Can we move this `!!header[0].key` check down?  It's here and above and it seems worthy of a separate function just for complicated tables or some other mechanism to avoid accidentally removing this logic.